subprojects/core/src/main/java/org/gradle/initialization/exception/DefaultExceptionAnalyser.java

```diff
@@ -98,7 +103,13 @@ private Throwable findDeepestRootException(Throwable exception) {
         Throwable locationAware = null;
         Throwable result = null;
         Throwable contextMatch = null;
-        for (Throwable current = exception; current != null; current = current.getCause()) {
+        // Guard against malicious overrides of Throwable.equals by
+        // using a Set with identity equality semantics.
+        Set<Throwable> dejaVu = Collections.newSetFromMap(new IdentityHashMap<>());
```

Comment:
the JDK code starts by adding the current Throwable to the `dejaVu` set. It's not so important, but I guess we should too.

Reply:
doh, your code does it too 🤦